operators_sample_website/src/main/java/com/generalbytes/batm/server/extensions/website/RestServiceWebsite.java

```diff
@@ -0,0 +1,199 @@
+                return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity("{ \"error\": \"Missing parameter transaction_id\" }").build();
+            }
+            return SellExtensions.getExtensionContext().findTransactionByTransactionId(transactionId).getStatus();
+        } catch (AuthenticationException e) {
+            return responseInvalidApiKey();
+        } catch (Throwable e) {
+            log.error("Error - status", e);
+        }
+        return responseInternalServerError();
+    }
+
+    /**
+     * https://localhost:7743/extensions/website/terminals
+     *
+     * @param apiKey Morphis API key
+     * @return list of active terminals ( active = pingDelay < 5 min ).
+     */
+    @GET
+    @Path("/terminals")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object terminals(@HeaderParam("X-Api-Key") String apiKey) {
+
+        try {
+            IApiAccess iApiAccess = checkSecurity(apiKey);
+            return getTerminalsByApiKey(iApiAccess);
+        } catch (AuthenticationException e) {
+            return responseInvalidApiKey();
+        } catch (Throwable e) {
+            log.error("Error - terminals", e);
+        }
+        return responseInternalServerError();
+    }
+
+    /**
+     * Method helps collect terminals with a same Morphis API Access in CAS
+     *
+     * @param iApiAccess - Morphis API key
+     * @return List<ITerminal>
+     */
+    private List<ITerminal> getTerminalsByApiKey(IApiAccess iApiAccess) {
+        Collection<String> terminals = iApiAccess.getTerminalSerialNumbers();
+        List<ITerminal> filteredTerminals = new ArrayList<>();
+        terminals.forEach(terminalSerial -> {
+            ITerminal terminal = SellExtensions.getExtensionContext().findTerminalBySerialNumber(terminalSerial);
+            if (isOnline(terminal)) {
+                filteredTerminals.add(terminal);
+            }
+        });
+        return filteredTerminals;
+    }
+
+    /**
+     * Method filters terminals with long ping delay
+     *
+     * @param terminal - Iterminal terminal
+     * @return boolean
+     */
+    private boolean isOnline(ITerminal terminal) {
+        long now = System.currentTimeMillis();
+        return terminal.getLastPingAt() != null
+            && (terminal.getLastPingAt().getTime() + pingDelay) > now;
+    }
+
+    /**
+     * @param apiKey - Morphis API key
+     * @return IApiAccess - Authenticated API key
+     */
+    private IApiAccess checkSecurity(String apiKey) throws AuthenticationException {
+        IApiAccess iApiAccess = SellExtensions.getExtensionContext().getAPIAccessByKey(apiKey);
+        if (iApiAccess == null) {
+            throw new AuthenticationException("Authentication failed");
+        }
+        return iApiAccess;
+    }
+
+    private Response responseInvalidParameter(String paramName) {
+        return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity("{ \"error\": \"Parameter " + paramName + " can't be null\"}").build();
```

Comment:
it would be nicer to have some `ErrorResponse` object with `String error` field (or at least a Map<String, String>) instead of constructing the json manually